nixos/modules/services/databases/neo4j.nix

```diff
@@ -122,18 +508,27 @@ in {
       after = [ "network.target" ];
       environment = {
         NEO4J_HOME = "${cfg.package}/share/neo4j";
-        NEO4J_CONF = "${cfg.dataDir}/conf";
+        NEO4J_CONF = "${toString cfg.directories.home}/conf";
       };
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/neo4j console";
         User = "neo4j";
         PermissionsStartOnly = true;
         LimitNOFILE = 40000;
       };
+
       preStart = ''
-        mkdir -m 0700 -p ${cfg.dataDir}/{data/graph.db,conf,logs}
-        ln -fs ${serverConfig} ${cfg.dataDir}/conf/neo4j.conf
-        if [ "$(id -u)" = 0 ]; then chown -R neo4j ${cfg.dataDir}; fi
+        # Directories Setup
+        mkdir -m 0700 -p ${toString cfg.directories.home}/{certificates,conf,data,import,logs,plugins}
+
+        # SSL Policies setup
+        ${concatStringsSep "\n" (map (p: ''
+          mkdir -m 0700 -p ${toString cfg.directories.home}/certificates/${p}/{revoked,trusted}
+        '') (builtins.attrNames cfg.ssl.policies))}
+
+        # Place the configuration where Neo4j can find it.
+        ln -fs ${serverConfig} ${toString cfg.directories.home}/conf/neo4j.conf
+        if [ "$(id -u)" = 0 ]; then chown -R neo4j ${toString cfg.directories.home}; fi
```

Comment:
Isn't this condition always true?

Reply:
Looks like it. This check is in the existing module, and I hadn't looked closely at it, apart from updating the directory ref.